.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,203 @@
+name: run-tests
+on:
+  workflow_call:
```

Comment:
`workflow_call` lets us reuse this across all of the places where we might want to run tests, which is pretty helpful.